Change the default stance to Defend for human players

obrakmann opened the pull request:
Exceptions are dogs and stealthy things.

Closes #3285.

Thanks to @matija-hustic for laying the groundwork that makes this work properly in the first place!

obrakmann committed:
Change InitialStance to Defend for human players

obrakmann committed:
Add an ActorInit for unit stances

obrakmann committed:
Add testcase for StanceInit

abcdefg30 commented on mods/cnc/rules/vehicles.yaml:
```diff
@@ -151,6 +151,7 @@ ARTY:
 	WithMuzzleOverlay:
 	AutoTarget:
 		InitialStance: Defend
```
This can be removed then?

abcdefg30 commented on mods/cnc/rules/vehicles.yaml:
```diff
@@ -491,6 +492,7 @@ MLRS:
 		AmmoPoolName: primary
 	AutoTarget:
 		InitialStance: Defend
```
Same here.

pchote commented:
I'm still very wary about this, but we can always revert it if (when?) our non-pro players complain / file confused bug reports about broken unit AI.

obrakmann commented:
Updated. The date in the upgrade rules will probably need to be changed before merging.

pchote commented:
Adding to the milestone because if we don't take this before the first playtest then I will insist that we leave it to the next cycle.  Slipping this in later in the testing cycle is asking for trouble.

obrakmann commented:
Rebased onto bleed, also dropped the commit that set the `Playable` flag on the scripted players to `false`, since that's already the default.

pchote commented on OpenRA.Mods.Common/Traits/AutoTarget.cs:
```diff
@@ -24,8 +24,11 @@ public class AutoTargetInfo : ITraitInfo, Requires<AttackBaseInfo>
 
-		[Desc("Possible values are HoldFire, ReturnFire, Defend and AttackAnything.")]
-		public readonly UnitStance InitialStance = UnitStance.AttackAnything;
+		[Desc("Possible values are HoldFire, ReturnFire, Defend and AttackAnything. Used for computer-controlled players.")]
```
Could you clarify this a bit wrt mission players?

obrakmann replied:
Hmm, like this? Not sure what kind of clarification you're asking for. I updated the `Desc` and renamed the variable to `InitialStanceAI`.

pchote replied:
That's better, thanks.

pchote commented:
Code changes look good, but I haven't tested this so :+0.5:.

penev92 commented:
You missed fremen with the "stealthy things".

Other than that :+1:

obrakmann commented:
Added default stance for fremen. Also changed the stances of all stealthy units to "HoldFire" for human players and "ReturnFire" for the AI, as discussed in IRC.

Mailaender commented on mods/ra/maps/fort-lonestar/map.yaml:
```diff
@@ -654,7 +654,7 @@ Rules:
 			HP: 200
 		AutoTarget:
-			InitialStance: Defend
+			InitialBotStance: Defend
```
I guess this has to be `InitialStanceAI` to match the code although `InitialBotStance` has better readability.

obrakmann replied:
Well spotted! I only ran sed on the rules directory and forgot about this.

abcdefg30 commented:
Looks good to me, too. :+1: / :white_check_mark:

abcdefg30 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/9de9115f976ea6a64baeb94602ddde442fd785e7...8a09b406e165173328c7edc4c0f60de77828cc4a)
